### PR TITLE
Bluetooth: controller: split: Fix missing slave latency impl.

### LINKED ISSUE #19525: Can't change the slave latency on a connection.
Zephyr 2.0.0
 I can't seem to change the slave latency on a connection.  I am using the central_hr and peripheral_hr sample code.
After connecting, i execute the following:
   param = BT_LE_CONN_PARAM(40, 40, 2, 420);
   conn_update(shell, param);The result (captured on an Elisys) is:

    LE-C Packet (Connection Parameter Request, Min= 50.000 ms, Max= 50.000 ms, Lat= 2, T/o= 4.200 s, Per= No Prefer
<Empty LE Packet
    Empty LE Packet
<LE-C Packet (Connection Parameter Response, Min= 50.000 ms, Max= 50.000 ms, Lat= 2, T/o= 4.200 s, Per= No Prefe
    Empty LE Packet
<Empty LE Packet
    LE-C Packet (Connection Update Indication, WinSz= 1.250 ms, WinOfs= 0, Int= 50.000 ms, Lat= 2, T/o= 4.200 s, Inst=
<Empty LE Packet
    Empty LE Packet
<Empty LE Packet
    Empty LE Packet
<Empty LE Packet
    Empty LE Packet
<Empty LE Packet
    Empty LE Packet
<Empty LE Packet
    Empty LE Packet
<Empty LE Packet
    Empty LE Packet
<Empty LE PacketI expected the slave (<) to reply to every third master (>) packet.  Any ideas??

### PR DESCRIPTION
**Pull request by cvinayak**
Port the missing slave latency implementation from legacy
controller.

Fixes #19525.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ull_slave.c [hunk 1]
```diff
@@ -335,6 +337,13 @@ void ull_slave_done(struct node_rx_event_done *done, u32_t *ticks_drift_plus,
...
 					       preamble_to_addr_us);
 	}
+
+	if (conn->tx_head ||
+	    memq_peek(conn->lll.memq_tx.head, conn->lll.memq_tx.tail, NULL)) {
+		return 0;
+	}
+
+	return conn->lll.latency;
 }
```
Comment (mtpr-ot) [inline subsys/bluetooth/controller/ll_sw/ull_slave.c]: There really seems to be no advantage in packing the latency decision inside this function, as it doesn't rely on the input parameters or "extract timing from done". Moving the 3-4 lines to ull_conn.c would then contain the change in one file.

File: subsys/bluetooth/controller/ll_sw/ull_conn.c [hunk 2]
```diff
@@ -945,13 +945,20 @@ void ull_conn_done(struct node_rx_event_done *done)
...
+	 */
 	ticks_drift_plus = 0U;
 	ticks_drift_minus = 0U;
 	if (done->extra.trx_cnt) {
 		if (IS_ENABLED(CONFIG_BT_PERIPHERAL) && lll->role) {
-			ull_slave_done(done, &ticks_drift_plus,
-				       &ticks_drift_minus);
+			lll->latency_event = ull_slave_done(conn, done,
+							    &ticks_drift_plus,
+							    &ticks_drift_minus);
```
Comment (mtpr-ot) [inline subsys/bluetooth/controller/ll_sw/ull_conn.c]: Perhaps move latency code here from ull_slave_done

File: subsys/bluetooth/controller/ll_sw/ull_slave.c (hunk 1 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_slave.c]: Agreed. I did have the same thought, but decided, being a slave feature to keep it in ull_slave. I will move it into ull_conn.c.

**Review by mtpr-ot — APPROVED** [review]
LGTM

**Comment by cvinayak** [comment]
@carlescufi this can be merged

**Review by carlescufi — APPROVED** [review]

**Pull request merged**